src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -512,6 +549,18 @@ impl<'a, 'tcx> DocFolder for LinkCollector<'a, 'tcx> {
                 let base_node =
                     if item.is_mod() && item.attrs.inner_docs { None } else { parent_node };
 
+                // replace `Self` with suitable item's name
+                let resolved_self = if path_str.starts_with("Self::") {
+                    if let Some(ref name) = parent_name {
+                        path_str.replace("Self::", &format!("{}::", name))
+                    } else {
+                        path_str.to_string()
+                    }
+                } else {
+                    path_str.to_string()
+                };
+                let path_str = &resolved_self;
```

Comment:
```suggestion
                let resolved_self;
                if let Some(remain) = path_str.strip_prefix("Self::") {
                    if let Some(ref name) = parent_name {
                        resolved_self = format!("{}::{}", name, remain),
                        path_str = &resolved_self;              
                    }
                }
```

Reply:
```
error[E0658]: use of unstable library feature 'str_strip': newly added
   --> src/librustdoc/passes/collect_intra_doc_links.rs:554:48
    |
554 |                 if let Some(remain) = path_str.strip_prefix("Self::") {
    |                                                ^^^^^^^^^^^^
    |
    = note: see issue #67302 <https://github.com/rust-lang/rust/issues/67302> for more information
    = help: add `#![feature(str_strip)]` to the crate attributes to enable
```

Reply:
You need to add the feature attr. It is okay to do so in std.

Reply:
Or maybe
```
                if path_str.starts_with("Self::") {
                    if let Some(ref name) = parent_name {
                        resolved_self = path_str.replace("Self::", &format!("{}::", name));
                        path_str = &resolved_self;
                    }
                }
```
i can do both way so up to you

Reply:
It is not up to me but to authoritative reviewers. I'd use unstable feature because I see it nicer.

Reply:
Sure. I'm waiting then for a suggestion.

Reply:
Actually, who do you mean by "authoritative reviewers"?

Reply:
Rustdoc reviewers. Sorry I am not a native speaker.

Reply:
You mean e.g @GuillaumeGomez ? Nevermind, i'll wait more.


> Sorry I am not a native speaker.

Who cares. I mean, I like both non-native and native speakers. :)

Reply:
used `path_str.strip_prefix("Self::")` so resolving..